tests/visualization_tests/matplotlib_tests/test_slice.py

```diff
@@ -21,33 +23,47 @@ def test_plot_slice() -> None:
     # Test with no trial.
     study = prepare_study_with_trials(no_trials=True)
     figure = plot_slice(study)
-    assert len(figure.get_lines()) == 0
+    assert len(figure.findobj(PathCollection)) == 0
 
     study = prepare_study_with_trials(with_c_d=False)
 
     # Test with a trial.
     figure = plot_slice(study)
     assert len(figure) == 2
-    assert len(figure[0].get_lines()) == 0
-    assert len(figure[1].get_lines()) == 0
+    assert len(figure[0].findobj(PathCollection)) == 1
+    assert len(figure[1].findobj(PathCollection)) == 1
     assert figure[0].yaxis.label.get_text() == "Objective Value"
 
+    # Scatter plot data is available as PathCollection
+    data0 = figure[0].findobj(PathCollection)[0].get_offsets().data
+    data1 = figure[1].findobj(PathCollection)[0].get_offsets().data
+    assert np.allclose(data0, [[1.0, 0.0], [2.5, 1.0]])
+    assert np.allclose(data1, [[2.0, 0.0], [0.0, 2.0], [1.0, 1.0]])
+
     # Test with a trial to select parameter.
     figure = plot_slice(study, params=["param_a"])
-    assert len(figure.get_lines()) == 0
+    assert len(figure.findobj(PathCollection)) == 1
     assert figure.yaxis.label.get_text() == "Objective Value"
 
+    data0 = figure.findobj(PathCollection)[0].get_offsets().data
+    assert np.allclose(data0, [[1.0, 0.0], [2.5, 1.0]])
```

Comment:
This pattern appears very frequently in this PR. To avoid code duplication, how about defining a function like:

```python
def _assert_data_allclose(figure, values):
    data = figure.findobj(PathCollection)[0].get_offsets().data
    assert np.allclose(data, values)
```

Reply:
I think we can leave them because tests for plotly also have the same point. If we introduce this helper, it would be nice to create another PR to update both tests.